agent/structs/connect_ca.go

```diff
@@ -133,6 +133,22 @@ func (c *CARoot) Clone() *CARoot {
 	return &newCopy
 }
 
+// LeafSigningCert returns the PEM encoded certificate that should be used to
+// sign leaf certificates in the local datacenter. The SubjectKeyId of the
+// returned cert should always match the SigningKeyID of the CARoot.
+//
+// This method has no knowledge of the provider, so it makes assumptions about
+// which cert is used based on established convention.
```

Comment:
We are already using this convention in metrics. I think it's generally correct, but I think it could be wrong in the case where we have a cross-signed cert in the `InteremdiateCerts` list without any local signing cert.

I'm not sure how we fix this without changing the `CARoot` struct to use different fields for these certs. I think maybe we should do that now.

Reply:
> We are already using this convention in metrics

For my learning 🤓 , what did you mean by this?

Reply:
Ah yes, I was referring to this code: https://github.com/hashicorp/consul/blob/v1.11.0-beta3/agent/consul/leader_metrics.go#L83-L98

Dhia added this in 1.11, so we do already use this logic, but it is after we have checked the provider to see if the provider uses an intermediate, so it's a bit safer there.  I need to double check if I need the same here.

Reply:
Looking at this again I'm a little uncomfortable exposing this as a method on the struct, so I think Ill move this to be an unexported function near where it is used instead, and improve the godoc to say that the provider should be checked first.

Reply:
Ok, the latest commit cleans this up and makes it safe to use without relying on these assumptions.